drivers/sensor/ak8975/ak8975.h

```diff
@@ -23,22 +23,20 @@
 #define AK8975_MEASURE_TIME_US		9000
 #define AK8975_MICRO_GAUSS_PER_BIT	3000
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#if DT_INST_0_ASAHI_KASEI_AK8975_BASE_ADDRESS != 0x0C
+#if DT_NODE_HAS_PROP(DT_INST(0, invensense_mpu9150), reg)
+#if DT_REG_ADDR(DT_INST(0, asahi_kasei_ak8975)) != 0x0C
 #error "I2C address must be 0x0C when AK8975 is part of a MPU9150 chip"
 #endif
 
-#ifdef DT_INST_0_INVENSENSE_MPU9150
-#define MPU9150_I2C_ADDR		DT_INST_0_INVENSENSE_MPU9150_BASE_ADDRESS
-#endif
+#define MPU9150_I2C_ADDR		DT_REG_ADDR(DT_INST(0, invensense_mpu9150))
```

Comment:
I was expecting `DT_INST_REG_ADDR(0)` (two places, if we're following driver idiom for this header), and this isn't going to be right if the instance ordinal for `invensense,mpu9150` isn't the same as the instance ordinal for `asahi-kasei,ak8975` (#20287) but I guess this is generally equivalent.